### PR TITLE
3.0.0: Dart 2 Support

### LINKED ISSUE #22: Dart 2 issue
Hello, 

after updating to dart 2. I get the following error when using `Reducer`
 

> error is happend type '(OnBoardingState, LicensesLoadedAction) => OnBoardingState' is not a subtype of type '(OnBoardingState, dynamic) => OnBoardingState'

My code looks like the following: 
```
final onBoardingReducer = combineTypedReducers<OnBoardingState>([
  new ReducerBinding <OnBoardingState, LicensesLoadedAction>(_setLoadedLicenses),
]);

OnBoardingState _setLoadedLicenses(OnBoardingState onBoardingData, LicensesLoadedAction action) {
  return onBoardingData.copyWith(licenses: action.licenses);
}

```

If I set the type to `dynamic` it works but I would prefer to not have to case the action. 

```
OnBoardingState _setLoadedSchools(OnBoardingState onBoardingData, dynamic action) {
  if(action is SchoolsLoadedAction) {
    return onBoardingData.copyWith(schools: action.schools);
  }
  return onBoardingData;
}
```

Any idea?

### PR DESCRIPTION
**Pull request by brianegan**
We were having some type issues with Dart 2, and I found that we needed to change the implementation of Redux a bit. Overall, I actually think there might be an underlying bug with Dart 2, but this change simplifies the way our typed combinations work and provides good support for Dart 2.

Fixes #22 

Changes:

  * Remove `ReducerBinding`, use `TypedReducer` 
  * Remove `combineTypedReducer`. Use `combineReducers` with normal reducers & `TypedReducer`s.
  * Remove `MiddlewareBinding`, use `TypedMiddleware`.
  * Remove `combineTypedMiddleware` -- no longer needed! Just create a normal `List<Middleware<State>>`!
  * Update docs for all of these cases

### CODE REVIEW

File: ios/Flutter/Generated.xcconfig [hunk 1]
```diff
@@ -0,0 +1,8 @@
+// This is a generated file; do not edit or check into version control.
+FLUTTER_ROOT=/Users/phillywiggins/lab/flutter
+FLUTTER_APPLICATION_PATH=/Users/phillywiggins/lab/redux.dart
+FLUTTER_TARGET=lib/main.dart
+FLUTTER_BUILD_MODE=debug
+FLUTTER_BUILD_DIR=build
+SYMROOT=${SOURCE_ROOT}/../build/ios
+FLUTTER_FRAMEWORK_DIR=/Users/phillywiggins/lab/flutter/bin/cache/artifacts/engine/ios
```
Comment (johnpryan) [inline ios/Flutter/Generated.xcconfig]: not sure why flutter generates these guys, let's leave them out

File: .travis.yml [hunk 2]
```diff
@@ -7,4 +7,3 @@ with_content_shell: false
 dart_task:
 - test: --platform vm
 - dartanalyzer: --fatal-warnings lib
-- dartfmt
```
Comment (johnpryan) [inline .travis.yml:10]: I've noticed that the Dart 1 and Dart 2 formatters format slightly differently. Is that why this was taken out?

**Review by johnpryan — COMMENTED** [review]
Look good overall - just a question and a nitpick

> Remove combineTypedMiddleware -- no longer needed! Just create a normal List<Middleware<State>>

This is a nice improvement!

File: ios/Flutter/Generated.xcconfig (hunk 1 above)
Comment (brianegan) [reply ios/Flutter/Generated.xcconfig]: Ack, good call. Will do.

File: .travis.yml (hunk 2 above)
Comment (brianegan) [reply .travis.yml:10]: Yep, no way I could satisfy both for the time being. Would like to add it back in eventually.

**Comment by brianegan** [comment]
Appreciate the review! I'll remove those extra files and publish a new version tomorrow morning :)

**Pull request merged**